packages/core/utils/universalEnv.js

```diff
@@ -1,5 +1,25 @@
-const getEnv = () => (
-  Object.keys(process.env).length ? process.env : window.__ENV__ // eslint-disable-line no-underscore-dangle
-);
+/* eslint-disable  no-underscore-dangle */
+const getEnv = () => {
+  const env = Object.keys(process.env).length ? process.env : window.__ENV__;
+
+  // If the `MULTISITE_CONTEXT` property has been added to window.__ENV__ during
+  // the webpack build process, replace instances of ROOT_URL and API_ROOT_URL
+  // with the corresponding values in the multisite configuration.
+  if (env.MULTISITE_CONTEXT) {
+    const { host } = window.location;
```

Comment:
Will this work server-side because of `shimDom`?

Reply:
I believe so. I have confirmed that the page source on a server-side render does include the correct pre-loaded state.